sqlite/trades/trades.go

```diff
@@ -0,0 +1,124 @@
+// Package trades provides an SQLite based trades database.
+package trades
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+)
+
+const (
+	insertSQL = `
+INSERT INTO trades (
+	time, symbol, price, buy
+) VALUES (
+	?, ?, ?, ?
+)
+`
+
+	schemaSQL = `
+CREATE TABLE IF NOT EXISTS trades (
+    time TIMESTAMP,
+    symbol VARCHAR(32),
+    price FLOAT,
+    buy BOOLEAN
+);
+
+CREATE INDEX IF NOT EXISTS trades_time ON trades(time);
+CREATE INDEX IF NOT EXISTS trades_symbol ON trades(symbol);
+`
+)
+
+// Trade is a buy/sell trade for symbol.
+type Trade struct {
+	Time   time.Time
+	Symbol string
+	Price  float64
+	IsBuy  bool
+}
+
+// DB is a database of stock trades.
+type DB struct {
+	sql    *sql.DB
+	stmt   *sql.Stmt
+	buffer []Trade
+}
+
+// NewDB constructs a Trades value for managing stock trades in a
+// SQLite database. This API is not thread safe.
+func NewDB(dbFile string) (*DB, error) {
+	sqlDB, err := sql.Open("sqlite3", dbFile)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err = sqlDB.Exec(schemaSQL); err != nil {
+		return nil, err
+	}
+
+	stmt, err := sqlDB.Prepare(insertSQL)
+	if err != nil {
+		return nil, err
+	}
+
+	db := DB{
+		sql:    sqlDB,
+		stmt:   stmt,
+		buffer: make([]Trade, 0, 1024),
+	}
+	return &db, nil
+}
+
+// Close flushes all trades to the database and prevents any future trading.
+func (db *DB) Close() (err error) {
+	defer func() {
+		err = db.sql.Close()
+	}()
+
+	if err := db.Flush(); err != nil {
+		return err
```

Comment:
If there's a error in `Flush` db.stmt` won't be closed. Do line 83 in a `defer` as well?